### ISSUE TITLE
Error: failed to retrieve kubeconfig

### ISSUE THREAD

**Issue by JasperZ** [siderolabs/terraform-provider-talos#88]
Today I did some testing with the updated terraform provider (v0.2.0). Everything except the data source **talos_cluster_kubeconfig** seems to work as excpected. Whenever I have the following code in my terraform file:
```
data "talos_cluster_kubeconfig" "this" {
  client_configuration = talos_machine_secrets.secrets.client_configuration
  endpoint             = local.cluster_talos_api_dns_name
  node                 = [for k, v in openstack_networking_port_v2.controlplane : v.all_fixed_ips[0]][0]

  wait = true

  depends_on = [
    talos_machine_bootstrap.bootstrap
  ]
}
```
I get this error:
```
│ Error: failed to retrieve kubeconfig
│
│   with module.cluster01.data.talos_cluster_kubeconfig.this,
│   on ../modules/openstack_talos_kubernetes_cluster/talos.tf line 103, in data "talos_cluster_kubeconfig" "this":
│  103: data "talos_cluster_kubeconfig" "this" {
│
│ empty retryable error received. This is a bug with the Terraform provider and should be reported as a GitHub issue in the provider repository.
```

The example code from the [talos_cluster_kubeconfig data source documentation](https://registry.terraform.io/providers/siderolabs/talos/latest/docs/data-sources/cluster_kubeconfig#example-usage) throws the same error.

I already experimented with the wait and timeout settings, but it didn't change anything. Using the provider version v0.2.0-beta.0 yields the same error. 
Downloading the kubeconfig via talosctl works as expected!

OS: Ubuntu 20.04 (WSL2)
Terraform Version: v1.4.6 (also tested with 1.4.2)
Talos Terraform Provider: v0.2.0 (also tested with v0.2.0-beta.0)
Talos OS: v1.4.1 (also tested with 1.3.7)

**Comment by frezbo** [comment]
Thanks for the bug report, this is already being tracked in #86

**Comment by dfsdevops** [comment]
I'm having the same issue, but I think it is because bootstrap is failing, however, I feel that bootstrap resource shouldn't be created until some kind of check is performed that core services are able to start. Maybe thats separate from what you are seeing.

**Comment by paucampana** [comment]
I am having the same issue.

**Comment by edgrz** [comment]
Hi,

I'm also facing the issue. Do you have a ball park estimation on when this will be fixed?

**Comment by frezbo** [comment]
Closed by #91
